src/02-room-details.js

```diff
@@ -49,7 +91,34 @@ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
       "Kit Hopkins Education Wing"
     ]
  */
-function getConnectedRoomNamesById(rooms, id) {}
+
+// This function will determine the name of the room connected to the given room
+// Declare a variable and assign it to an empty array
+// Create a for loop that loops through each room
+// IF the room ID cannot be found, return an error message
+// Refer to the test
+
+function getConnectedRoomNamesById(rooms, id) {
+  let emptyArr = [];
+  for (let i = 0; i < rooms.length; i++) {
+    const room = rooms[i];
+    if (room.roomId.includes(id)) {
+      emptyArr = room.connectsTo;
+    }
+    for (let a = 0; a < rooms.length; a++) {
+      const string = rooms[a];
+      if (string.connectsTo.includes(string.roomId)) {
+        emptyArr = string.name;
+      } else if (id !== string.roomId) {
```

Comment:
Because this is inside a for loop the second your code hits that this is true, it will conclude your entire function and not continue to iterate.